ggshield/oauth.py

```diff
@@ -0,0 +1,273 @@
+import os
+import urllib.parse as urlparse
+import webbrowser
+from base64 import urlsafe_b64encode
+from datetime import datetime
+from hashlib import sha256
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from secrets import token_urlsafe
+from typing import Any, Dict, Optional, Type
+
+import click
+import requests
+from oauthlib.oauth2 import OAuth2Error, WebApplicationClient
+
+from .client import retrieve_client
+from .config import AccountConfig, Config
+
+
+CLIENT_ID = "ggshield_oauth"
+SCOPE = "scan"
+REDIRECT_PORT = 1234
```

Comment:
we should look for an open port, and optionally to allow the user to choose the port manually.